tests/security_tests.py

```diff
@@ -28,8 +30,65 @@ def get_perm_tuples(role_name):
     return perm_set
 
 
+SCHEMA_ACCESS_ROLE = "schema_access_role"
+
+
 class RolePermissionTests(SupersetTestCase):
-    """Testing export import functionality for dashboards"""
+    """Testing export role permissions."""
+
+    def test_sqllab_gamma_user_access_to_sqllab(self):
+        session = db.session
+        security_manager.add_role(SCHEMA_ACCESS_ROLE)
+        session.commit()
+
+        example_db = session.query(Database).filter_by(database_name="examples").one()
+        example_db.expose_in_sqllab = True
+        session.commit()
+
+        ds = (
+            db.session.query(SqlaTable)
+            .filter_by(table_name="wb_health_population")
+            .first()
+        )
+        ds.schema = "temp_schema"
+        security_manager.add_permission_view_menu("schema_access", ds.schema_perm)
+        schema_perm_view = security_manager.find_permission_view_menu(
+            "schema_access", ds.schema_perm
+        )
+        security_manager.add_permission_role(
+            security_manager.find_role(SCHEMA_ACCESS_ROLE), schema_perm_view
+        )
+        gamma_user = security_manager.find_user(username="gamma")
+        gamma_user.roles.append(security_manager.find_role(SCHEMA_ACCESS_ROLE))
+        session.commit()
+
+        OLD_FLASK_GET_SQL_DBS_REQUEST = (
+            "databaseasync/api/read?_flt_0_expose_in_sqllab=1&"
+            "_oc_DatabaseAsync=database_name&_od_DatabaseAsync=asc"
+        )
+        self.login(username="gamma")
+        databases_json = self.client.get(OLD_FLASK_GET_SQL_DBS_REQUEST).json
+        assert databases_json["count"] == 1
+
+        NEW_FLASK_GET_SQL_DBS_REQUEST = (
+            "/api/v1/database/?q=(keys:!(none),filters:!((col:expose_in_sqllab,opr:eq,value:!t)),"
+            "order_columns:database_name,order_direction:asc,page:0,page_size:-1)"
+        )
```

Comment:
To make the code more readable we could take advantage of Rison and user something like:
``` python
arguments = {
    "keys": ["none"],
    "filters":[
        {
            "col": "expose_in_lab",
            "opr": "eq",
            "value": true
        }
    ],
    "order_columns": "database_name"
    "order_direction": "asc"
    "page": 0
    "page_size": -1
}
NEW_FLASK_GET_SQL_DBS_REQUEST = f"/api/v1/database/?q={prison.dumps(arguments)}"
```